src/System.IO.Packaging/src/System/IO/Packaging/ZipWrappingStream.cs

```diff
@@ -88,25 +84,14 @@ public override void Flush()
             _baseStream.Flush();
         }
 
-        public override long Position
+        public override long Length
         {
             get
             {
-                return _baseStream.Position;
-            }
-            set
-            {
-                _baseStream.Position = value;
-            }
-        }
-
-        // For some strange reason, if the package FileAccess == Read,
-        // then can't get length from the stream.  Not supported.
-        // The workaround - write the stream to a memory stream, then return the length of the
-        // memory stream.
-        public override long Length
-        {
-            get {
+                // For some strange reason, if the package FileAccess == Read,
+                // then can't get length from the stream.  Not supported.
+                // The workaround - write the stream to a memory stream, then return the length of the
+                // memory stream.
                 if (_packageFileAccess == FileAccess.Read)
```

Comment:
>Do you think this is actually still a relevant workaround? Maybe this has been fixed?

It's certainly possible and worth testing. I don't like that we're cloning the entire stream just to get the length "for some strange reason" :)

>I can't build the tests - getting a FileNotFoundException for test data, so I can't benchmark exact usage figures

mind posting the error?